src/alda/lisp/events/note.clj

```diff
@@ -37,22 +37,30 @@
                                                  time-scaling
                                                  ms)
           quant-duration     (* full-duration quant)
-          midi-note          (if (= event-type :note)
+          note-event?        (= event-type :note)
+          midi-note-val      (when note-event?
                                (determine-midi-note event octave key-signature
                                                     transposition))
-          pitch              (if (= event-type :note)
+          midi-note          (if (or (nil? midi-note-val) (<= 0 midi-note-val 127))
+                               midi-note-val
```

Comment:
Essentially, what we're doing here is setting up a binding `midi-note`, which will either end up being the same value as `midi-note-val` or it will throw an error.

I think we could slightly simplify this code by calling the calculation on lines 41-42 `midi-note` instead of `midi-note-val`, and then inserting a let-binding pair where we just throw an error if appropriate:

```clojure
midi-note (when note-event?
            (determine-midi-note ...)
_         (when (and midi-note (not (<= 0 midi-note 127)))
            (throw ...))
```